Common/src/Storage/ResultLifeCycleHelper.cs

```diff
@@ -35,31 +35,34 @@ namespace ArmoniK.Core.Common.Storage;
-  ///   Abort all tasks and results that depend directly or indirectly of the specified tasks
+  ///   Terminate all tasks and results that depend directly or indirectly of the specified tasks.
+  ///   Tasks are set to an error status (or deleted if requested), and their results are marked as aborted.
   /// </summary>
   /// <param name="taskTable">Interface to manage task states</param>
   /// <param name="resultTable">Interface to manage result states</param>
-  /// <param name="taskIds">Root tasks that must be aborted</param>
-  /// <param name="resultIds">Root results that must be aborted</param>
-  /// <param name="reason">Abortion message</param>
-  /// <param name="status">Put aborted tasks into the given status</param>
+  /// <param name="taskIds">Root tasks that must be terminated</param>
+  /// <param name="resultIds">Root results that must be terminated</param>
+  /// <param name="reason">Termination message</param>
+  /// <param name="status">Put terminated tasks into the given status</param>
+  /// <param name="shouldDeleteTasks">Should delete tasks from database</param>
   /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
   /// <returns>
   ///   Task representing the asynchronous execution of the method
   /// </returns>
-  public static async Task AbortTasksAndResults(ITaskTable           taskTable,
-                                                IResultTable         resultTable,
-                                                ICollection<string>? taskIds           = null,
-                                                ICollection<string>? resultIds         = null,
-                                                string?              reason            = null,
-                                                TaskStatus           status            = TaskStatus.Error,
-                                                CancellationToken    cancellationToken = default)
+  public static async Task TerminateTasksAndResults(ITaskTable taskTable,
+                                                     IResultTable resultTable,
+                                                     ICollection<string>? taskIds = null,
+                                                     ICollection<string>? resultIds = null,
+                                                     string? reason = null,
+                                                     TaskStatus status = TaskStatus.Error,
+                                                     bool shouldDeleteTasks = false,
+                                                     CancellationToken cancellationToken = default)
   {
-    taskIds   ??= Array.Empty<string>();
+    taskIds ??= Array.Empty<string>();
     resultIds ??= Array.Empty<string>();
 
-    // Early exit if no abortion is actually requested
-    if (!taskIds.Any() && !resultIds.Any())
+    // Early exit if no termination is actually requested
+    if (taskIds.Count == 0 && resultIds.Count == 0)
```

Comment:
I'd rather not change this condition as we prefer to use this style overall.
Formatting and lexical analysis will potentially make you change it.